src/main/scala/nl.knaw.dans.easy.solr/SolrDocumentGenerator.scala

```diff
@@ -216,6 +228,18 @@ abstract class SolrDocumentGenerator(pid: String) extends DebugEnhancedLogging {
     }
   }
 
+  implicit private def dateTimeOrdering: Ordering[DateTime] = Ordering.fromLessThan(_ isBefore _)
+
+  def getAmdDateElement(typeOfDate: String): NodeSeq = {
+    val dates = (amd \ "stateChangeDates" \ "stateChangeDate")
+      .collect { case change if (change \ "toState").text.toLowerCase == typeOfDate => (change \ "changeDate").text }
+
+    if (dates.isEmpty)
+      NodeSeq.Empty
+    else
+      <node>{dates.maxBy(DateTime.parse)}</node>
```

Comment:
Is this `maxBy` here so that a the latest value will be selected if a dataset was published more than once for example? If so, may add a unit test for that?

Reply:
your observation is correct. [But we already have such a test for the 'SUBMITTED' case rather than 'PUBLISHED'.](https://github.com/DANS-KNAW/easy-update-solr-index/blob/9fc3bf2dff2ea91f758461bffe880f86f4bdfaa6/src/test/scala/nl.knaw.dans.easy.solr/SolrDocumentGeneratorSpec.scala#L76) Hence I don't see the need for yet another test.